pyomo/solvers/plugins/solvers/gurobi_direct.py

```diff
@@ -288,7 +292,11 @@ def _add_sos_constraint(self, con):
 
         self._vars_referenced_by_con[con] = ComponentSet()
 
-        for v, w in con.get_items():
+        if hasattr(con, 'get_items'):
+            sos_items = con.get_items()
+        else:
+            sos_items = con.items()
+        for v, w in sos_items:
```

Comment:
Same question as above: why not use `six.iteritems(con)`?

Reply:
See Gabe's response to the question above.